mesos-swarm-marathon/configure-mesos-cluster.sh

```diff
@@ -109,77 +199,91 @@ if ismaster ; then
   done
 fi
 
-#########################
-# Configure Mesos Master
-#########################
+#########################################
+# Configure Mesos Master and Frameworks
+#########################################
 if ismaster ; then
   quorum=`expr $MASTERCOUNT / 2 + 1`
   echo $quorum | sudo tee /etc/mesos-master/quorum
   hostname -i | sudo tee /etc/mesos-master/ip
   hostname | sudo tee /etc/mesos-master/hostname
+  echo 'Mesos Cluster on Microsoft Azure' | sudo tee /etc/mesos-master/cluster
+fi
+
+if ismaster  && [ "$MARATHONENABLED" == "true" ] ; then
+  # setup marathon
   sudo mkdir -p /etc/marathon/conf
   sudo cp /etc/mesos-master/hostname /etc/marathon/conf
   sudo cp /etc/mesos/zk /etc/marathon/conf/master
   zkmarathonconfig=$(zkconfig "marathon")
   echo $zkmarathonconfig | sudo tee /etc/marathon/conf/zk
-  echo 'Mesos Cluster on Microsoft Azure' | sudo tee /etc/mesos-master/cluster
 fi
 
 #########################
 # Configure Mesos Agent
 #########################
 if isagent ; then
+  # Add docker containerizer
+  echo "docker,mesos" | sudo tee /etc/mesos-slave/containerizers
   hostname -i | sudo tee /etc/mesos-slave/ip
   hostname | sudo tee /etc/mesos-slave/hostname
 fi
 
-################
-# Install Docker
-################
-
-echo "Installing and configuring docker and swarm"
-
-time wget -qO- https://get.docker.com | sh
-
-if isagent ; then
-  # Start Docker and listen on :2375 (no auth, but in vnet)
-  echo 'DOCKER_OPTS="-H unix:// -H 0.0.0.0:2375"' | sudo tee /etc/default/docker
-  sudo service docker restart
-fi
-
-# Run swarm manager container on port 2376 (no auth)
-if ismaster ; then
-  time sudo docker run -d -e SWARM_MESOS_USER=root \
-      --restart=always \
-      -p 2376:2375 -p 3375:3375 swarm manage \
-      -c mesos-experimental \
-      --cluster-opt mesos.address=0.0.0.0 \
-      --cluster-opt mesos.port=3375 $masterVMName:5050
-fi
-
-echo "Finished installing and configuring docker and swarm"
-
 ##############################################
 # configure init rules restart all processes
 ##############################################
 
-echo "restarting mesos processes"
+echo "(re)starting mesos and framework processes"
 if ismaster ; then
   sudo restart zookeeper
   sudo start mesos-master
-  sudo start marathon
+  if [ "$MARATHONENABLED" == "true" ] ; then
+    sudo start marathon
```

Comment:
Not sure what the difference between "service xxx restart" vs "restart xxx" but prefer one for consistency..

I think service command abstracts out init system (upstart vs systemd) might be more portable. @anhowe